lib/collection/response.js

```diff
@@ -211,6 +211,30 @@ _.extend(Response.prototype, /** @lends Response.prototype */ {
         // we create the body string first from stream and then fallback to body
         return 'data:' + mime.type + ';base64, ' + (((this.stream != null) &&
             util.bufferOrArrayBufferToBase64(this.stream)) || ((this.body != null) && util.btoa(this.body)) || E);
+    },
+
+    /**
+     * Get the response size
+     * @return {Number}
+     * @todo write unit tests
+     */
+    size: function () {
+        var sizeInfo = {},
+            contentEncoding = Header.headerValue(this.headers, 'Content-Encoding'),
+            contentLength = Header.headerValue(this.headers, 'Content-Length');
+
+        // if 'Content-Length' header is present and encoding is of type gzip/deflate
+        if (contentLength && /^(gzip|deflate)$/.test(contentEncoding) && util.isNumeric(contentLength)) {
+            sizeInfo.bodySize = _.parseInt(contentLength, 10);
+        }
+        // else size of body is added
+        else {
+            sizeInfo.bodySize = this.stream ? this.stream.byteLength : (this.body && this.body.length || 0);
+        }
+
+        // size of header is added
+        sizeInfo.headerSize = Header.size(this.headers, this.code, this.reason());
+        return sizeInfo;
```

Comment:
Looks good except for the size info property names. 

You are returning a size, so it is redundant to make the internal property suffixed "size".

response.size().body and respose.size().header sounds more natural.